firmware/src/main.cpp

```diff
@@ -89,6 +89,8 @@ void loop() {
 
         MainHelper::checkCycleWidgets();
         wifiManager->process();
+        HTTPClientWrapper::getInstance()->processRequestQueue();  
+        HTTPClientWrapper::getInstance()->processResponseQueue();
```

Comment:
I'd combine those functions, that way you can guarantee the order of the handling and other things that might be important for various reasons.

Reply:
Combining them won't guarantee the order of handling since the processRequestQueue() generates a non-blocking Task thread.  You can't wait for the response or it would be blocking.